WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -322,10 +322,11 @@ protected SendControlViewModel(Wallet wallet, string title)
 
 					await BuildTransaction(Password, intent, feeStrategy, allowUnconfirmed: true, allowedInputs: selectedCoinReferences);
 				}
-				catch (InsufficientBalanceException ex)
+				catch (NotEnoughFundsException ex)
 				{
-					Money needed = ex.Minimum - ex.Actual;
-					NotificationHelpers.Error($"Not enough coins selected. You need an estimated {needed.ToString(false, true)} BTC more to make this transaction.", "");
+					// SmartCoinSelector.Select ran for multiple times by NBitcoin - for the first time ex.Minimum does not contain the fee.
+					Money needed = Amount - AllSelectedAmount;
```

Comment:
I don't like this. We set the amount in a completely different place. That's how future bugs are created. Not sure what should be the right thing though.

Reply:
Same for AllSelectedAmount. It is probably even be possible today that while NBitcoin is working the selection changes.